### ISSUE TITLE
MaterialBanner: Allow customizing the height of the banner

### ISSUE THREAD

**Issue by itsatifsiddiqui** [flutter/flutter#153666]
## Steps to Reproduce

1. Create a MaterialBanner widget
2. Attempt to set a custom `padding` of zero for lable
3. Observe that the banner cannot be smaller than 52 pixels in height

## Expected results

The MaterialBanner should respect padding, allowing for more compact designs when needed.

## Actual results

The MaterialBanner has a minimum height of 52 pixels due to hardcoded constraints, preventing the creation of more compact banners.

## Code sample

```dart
ScaffoldMessenger.of(context)
  ..hideCurrentMaterialBanner()
  ..showMaterialBanner(
    const MaterialBanner(
      backgroundColor: Colors.green,
      content: Text('Too much vertical space between the text'),
      padding: EdgeInsets.zero,
      leadingPadding: EdgeInsets.zero,
      actions: [
        SizedBox(),
      ],
    ),
  );
```

## Details

The MaterialBanner widget has a hardcoded minimum height of 52 pixels for its actions bar, which prevents creating more compact banners. This limitation is due to the following code in the MaterialBanner implementation:
```dart
final Widget actionsBar = Container(
  alignment: AlignmentDirectional.centerEnd,
  constraints: const BoxConstraints(minHeight: 52.0),
  padding: const EdgeInsets.symmetric(horizontal: 8),
  child: OverflowBar(
    overflowAlignment: widget.overflowAlignment,
    spacing: 8,
    children: widget.actions,
  ),
);
```

### Proposal

To address this limitation, we could:

Make the minHeight a customizable property of MaterialBanner.
Remove the hardcoded constraint and allow the banner to size based on its content.
Introduce a compact mode for MaterialBanner that uses smaller default values.

These changes would enhance the flexibility of MaterialBanner, allowing developers to create UIs that better match their design specifications, especially for apps requiring a more compact layout or needing to display banners in limited screen space.
Impact
This change would be a minor API addition (if adding a new property) or a behavioral change (if removing the constraint). It would improve the flexibility of MaterialBanner without breaking existing implementations that rely on the current minimum height.

**Comment by itsatifsiddiqui** [comment]
The below red banner is the desired behaviour I am trying to achieve. 

![Simulator Screenshot - iPhone 15 Pro - 2024-08-19 at 03 51 54](https://github.com/user-attachments/assets/881c201a-9965-4906-93af-659f5b756eae)

**Comment by piedcipher** [comment]
Fixes in #153951
